projects/compound/index.js

```diff
@@ -2,164 +2,295 @@
   Modules
   ==================================================*/
 
-  const sdk = require('../../sdk');
-  const _ = require('underscore');
-  const BigNumber = require('bignumber.js');
+const sdk = require('../../sdk');
+const _ = require('underscore');
+const BigNumber = require('bignumber.js');
 
-  const abi = require('./abi.json');
+const abi = require('./abi.json');
 
 /*==================================================
   TVL
   ==================================================*/
 
-  async function tvl(timestamp, block) {
-    let balances = {};
-
-    let tokens = (await sdk.api.cdp.compound.tokens()).output;
-
-    // V1
-
-    let balanceOfResults = await sdk.api.abi.multiCall({
-      block,
-      calls: _.map(tokens, (token, address) => ({
-        target: address,
-        params: '0x3FDA67f7583380E67ef93072294a7fAc882FD7E7'
-      })),
-      abi: 'erc20:balanceOf'
-    });
-
-    await sdk.util.sumMultiBalanceOf(balances, balanceOfResults);
-
-    // V2
-
-    let getCashResults = await sdk.api.abi.multiCall({
-      block,
-      calls: _.map(tokens, (token, address) => ({
-        target: token.cToken
-      })),
-      abi: {
-        "constant":true,
-        "inputs":[
-
-        ],
-        "name":"getCash",
-        "outputs":[
-          {
-            "name":"",
-            "type":"uint256"
-          }
-        ],
-        "payable":false,
-        "signature":"0x3b1d21a2",
-        "stateMutability":"view",
-        "type":"function"
-      }
-    });
-
-    _.each(tokens, (token, address) => {
-      let getCash = _.find(getCashResults.output, (result) => { return result.success && result.input.target == token.cToken});
+// ask comptroller for all markets array
+async function getAllCTokens(block) {
+  let res = await sdk.api.abi.call({
+    block,
+    target: '0x3d9819210a31b4961b30ef54be2aed79b9c9cd3b',
+    params: [],
+    abi: {
+      constant: true,
+      inputs: [],
+      name: 'getAllMarkets',
+      outputs: [
+        {
+          internalType: 'contract CToken[]',
+          name: '',
+          type: 'address[]',
+        },
+      ],
+      payable: false,
+      stateMutability: 'view',
+      type: 'function',
+      signature: '0xb0772d0b',
+    },
+  })
+  return res.output;
+}
+
+async function getUnderlying(block, cToken) {
+  if (cToken == '0x4Ddc2D193948926D02f9B1fE9e1daa0718270ED5') {
+    return '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2';//cETH => WETH
+  }
+  let res = await sdk.api.abi.call({
+    block,
+    target: cToken,
+    abi: {
+      constant: true,
+      inputs: [],
+      name: 'underlying',
+      outputs: [
+        {
+          name: '',
+          type: 'address',
+        },
+      ],
+      payable: false,
+      stateMutability: 'view',
+      type: 'function',
+      signature: '0x6f307dc3',
+    },
+  })
+  return res.output;
+}
+
+// returns {[underlying]: {cToken, decimals, symbol}}
+async function getMarkets(block) {
+  // cache some data
+  let markets = {
```

Comment:
Just wondering if we need this- it would be nice (albeit heavier) to simply pull the data each time by iterating over the tokens and pulling the necessary data, right? Just wondering how we prevent having to add this each time a new token is listed on Compound. Additionally, I'm not heavily against dropping data from v1 since it's now a small fraction of the overall markets.

Reply:
just seemed a little silly to pull the same data every hour. i think they want v1 data so they can have historical graphs

Reply:
Ah that makes sense. Looks good to me, overall, then!